components/service/pocket/src/test/java/mozilla/components/service/pocket/api/PocketEndpointTest.kt

```diff
@@ -0,0 +1,101 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.pocket.api
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import mozilla.components.concept.fetch.Client
+import mozilla.components.service.pocket.helpers.PocketTestResource
+import mozilla.components.service.pocket.helpers.TEST_ARTICLES_COUNT
+import mozilla.components.service.pocket.helpers.TEST_ARTICLES_LOCALE
+import mozilla.components.service.pocket.helpers.TEST_VALID_API_KEY
+import mozilla.components.service.pocket.helpers.assertClassVisibility
+import mozilla.components.service.pocket.helpers.assertResponseIsFailure
+import mozilla.components.support.test.any
+import mozilla.components.support.test.mock
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito
+import kotlin.reflect.KVisibility
+
+@RunWith(AndroidJUnit4::class)
+class PocketEndpointTest {
+
+    private lateinit var endpoint: PocketEndpoint
+    private lateinit var raw: PocketEndpointRaw // we shorten the name to avoid confusion with endpoint.
+    private lateinit var jsonParser: PocketJSONParser
+
+    private lateinit var client: Client
+
+    @Before
+    fun setUp() {
+        raw = mock()
+        jsonParser = mock()
+        endpoint = PocketEndpoint(raw, jsonParser)
+
+        client = mock()
+    }
+
+    @Test
+    fun `GIVEN a PocketEndpoint THEN its visibility is internal`() {
+        assertClassVisibility(PocketEndpoint::class, KVisibility.INTERNAL)
+    }
+
+    @Test
+    fun `WHEN getting articles recommendations and the endpoint returns null THEN a failure is returned`() {
+        Mockito.`when`(raw.getGlobalArticlesRecommandations(TEST_ARTICLES_COUNT, TEST_ARTICLES_LOCALE)).thenReturn(null)
```

Comment:
nit: import `Mockito.when`. You can also use `whenever` to avoid the back ticks.